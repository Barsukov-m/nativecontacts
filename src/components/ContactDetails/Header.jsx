import { Image, Text, View } from 'react-native';
import styles from './styles';

const Header = ({ title, image }) => {
  return (
    <View style={styles.header}>
      {image && <Image source={{ uri: image }} style={styles.image} />}
      <Text style={styles.contactName}>{title}</Text>
    </View>
  );
};

export default Header;
