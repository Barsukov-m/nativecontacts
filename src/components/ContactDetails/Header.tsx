import { Image, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import styles from './styles';

interface HeaderProps {
	title: string;
	image?: string;
}

const Header: React.FC<HeaderProps> = ({ title, image }) => {
	const renderHeaderImage = () => {
		if (image) {
			return (
				<Image
					source={{ uri: image }}
					style={styles.image}
					testID="contact-image"
				/>
			);
		} else {
			return (
				<Icon
					style={{ ...styles.image, ...styles.missingImage }}
					name="person"
					size={60}
				/>
			);
		}
	};
	return (
		<View style={styles.header}>
			{renderHeaderImage()}
			<Text style={styles.contactName}>{title}</Text>
		</View>
	);
};

export default Header;
