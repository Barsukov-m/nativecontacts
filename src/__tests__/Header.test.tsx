import { render } from '@testing-library/react-native';
import Header from '../components/ContactDetails/Header';

describe('<Header />', () => {
  it('renders title correctly', () => {
    const { getByText } = render(<Header title="Miss Marina Hauser" />);

    // Check for the contact name
    expect(getByText('Miss Marina Hauser')).toBeDefined();
  });

  it('renders image if provided', () => {
    const title = 'Miss Marina Hauser';
    const imageUri = 'https://randomuser.me/api/portraits/women/22.jpg';
    const testID = 'contact-image';

    const { getAllByTestId } = render(
      <Header title={title} image={imageUri} />
    );

    const imageComponents = getAllByTestId(testID);

    // Check for the contact profile image
    expect(imageComponents).toBeDefined();
  });
});
