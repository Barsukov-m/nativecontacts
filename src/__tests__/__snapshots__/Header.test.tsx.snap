// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders title correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <Text>
    Icon
  </Text>
  <Text
    style={
      {
        "color": "#222",
        "fontSize": 20,
        "marginVertical": 20,
      }
    }
  >
    Marina Hauser
  </Text>
</View>
`;
